client/state/imports/actions.js

```diff
@@ -0,0 +1,312 @@
+		importerId,
+		siteId,
+	} );
+	dispatch( { type: IMPORTS_FETCH } );
+	Dispatcher.handleViewAction( { type: IMPORTS_FETCH } );
+
+	wpcom
+		.updateImporter(
+			siteId,
+			toApi( {
+				importerId,
+				importerState: appStates.EXPIRE_PENDING,
+				site: { ID: siteId },
+			} )
+		)
+		.then( data => {
+			const importerStatus = fromApi( data );
+
+			dispatch( { type: IMPORTS_FETCH_COMPLETED } );
+			Dispatcher.handleViewAction( { type: IMPORTS_FETCH_COMPLETED } );
+			dispatch( {
+				type: IMPORTS_IMPORT_RECEIVE,
+				importerStatus,
+			} );
+			Dispatcher.handleViewAction( {
+				type: IMPORTS_IMPORT_RECEIVE,
+				importerStatus,
+			} );
+		} )
+		.catch( () => {
+			dispatch( { type: IMPORTS_FETCH_FAILED } );
+			Dispatcher.handleViewAction( { type: IMPORTS_FETCH_FAILED } );
+		} );
+};
+
+export const startImport = ( siteId, importerType ) => dispatch => {
+	const action = {
+		// Use a fake ID until the server returns the real one
+		importerId: `${ ID_GENERATOR_PREFIX }${ Math.round( Math.random() * 10000 ) }`,
+		type: IMPORTS_IMPORT_START,
+		importerType,
+		siteId,
+	};
+
+	dispatch( action );
+	Dispatcher.handleViewAction( action );
+};
+
+export const startImporting = importerStatus => {
+	const {
+		importerId,
+		site: { ID: siteId },
+	} = importerStatus;
+
+	return dispatch => {
+		dispatch( { type: IMPORTS_IMPORT_UNLOCK, importerId } );
+		Dispatcher.handleViewAction( { type: IMPORTS_IMPORT_UNLOCK, importerId } );
+		dispatch( {
+			type: IMPORTS_START_IMPORTING,
+			importerId,
+		} );
+		Dispatcher.handleViewAction( {
+			type: IMPORTS_START_IMPORTING,
+			importerId,
+		} );
+
+		wpcom.updateImporter(
+			siteId,
+			toApi( {
+				...importerStatus,
+				importerState: appStates.IMPORTING,
+			} )
+		);
+	};
+};
+
+export const finishUploadBasic = ( importerId, importerStatus ) => ( {
+	type: IMPORTS_UPLOAD_COMPLETED,
+	importerId,
+	importerStatus,
+} );
+
+export const finishUpload = ( importerId, importerStatus ) => dispatch => {
+	const action = finishUploadBasic( importerId, importerStatus );
+
+	dispatch( action );
+	Dispatcher.handleViewAction( action );
+};
+
+export const setUploadProgress = ( importerId, { uploadLoaded, uploadTotal } ) => dispatch => {
+	const action = {
+		type: IMPORTS_UPLOAD_SET_PROGRESS,
+		uploadLoaded,
+		uploadTotal,
+		importerId,
+	};
+
+	dispatch( action );
+	Dispatcher.handleViewAction( action );
+};
+
+export const failUpload = ( importerId, error ) => ( {
+	type: IMPORTS_UPLOAD_FAILED,
+	importerId,
+	error,
+} );
+
+export const startUpload = ( importerStatus, file ) => dispatch => {
+	const {
+		importerId,
+		site: { ID: siteId },
+	} = importerStatus;
+
+	wpcom
+		.uploadExportFile( siteId, {
+			importStatus: toApi( importerStatus ),
+			file,
+			onprogress: event =>
+				dispatch(
+					setUploadProgress( importerId, {
+						uploadLoaded: event.loaded,
+						uploadTotal: event.total,
+					} )
+				),
+
+			onabort: () => cancelImport( siteId, importerId ),
```

Comment:
An oversight here: cancelImport needs to be called with `dispatch`... right now this would just return a function. Working on fixing this up and looking for other similar instances